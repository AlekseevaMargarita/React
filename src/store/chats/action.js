
export const ADD_CHAT = 'CHATS::ADD_CHAT';
export const DELETE_CHAT = 'CHATS::DELETE_CHAT';

export const addChat = (chat) => ({
    type: ADD_CHAT,
    payload: {
        chat,
    },
});

export const deleteChat = (chatId) => ({
    type: DELETE_CHAT,
    payload: {
        chatId,
    }
});