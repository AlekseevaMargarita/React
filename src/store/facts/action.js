import { apiUrl } from "../../constants/constants";

export const GET_FACTS_REQUEST = "FACTS::GET_FACTS_REQUEST";
export const GET_FACTS_SUCCESS = "FACTS::GET_FACTS_SUCCESS";
export const GET_FACTS_FAILURE = "FACTS::GET_FACTS_FAILURE";

export const getFactsRequest = () => ({
    type: GET_FACTS_REQUEST,
});

export const getFactsSuccess = (facts) => ({
    type: GET_FACTS_SUCCESS,
    payload: facts,
});

export const getFactsFailure = (err) => ({
    type: GET_FACTS_FAILURE,
    payload: err,
});

export const getFacts = () => async (dispatch) => {
    dispatch(getFactsRequest());

    try {
        const response = await fetch(apiUrl);
        if (!response.ok) {
            throw new Error(`Request failed with status ${response.status}`);
        }
        const result = await response.json();
        dispatch(getFactsSuccess(result));
    }
    catch (err) {
        dispatch(getFactsFailure(err.message));
    }
};
