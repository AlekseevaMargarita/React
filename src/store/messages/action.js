/* import { AUTHORS } from "../../constants/constants"; */

export const ADD_MESSAGE = 'MESSAGES::ADD_MESSAGE';
export const UPDATE_MESSAGES = 'MESSAGES::UPDATE_MESSAGES';

export const addMessage = (chatId, message) => ({
    type: ADD_MESSAGE,
    payload: {
        chatId,
        message,
    },
});

/* let timerId;

export const addMessageWithThunk = (chatId, message) => (dispatch) => {
    dispatch(addMessage(chatId, message));

    if (timerId) {
        clearTimeout(timerId)
    }

    if (message.author !== AUTHORS.BOT) {
        timerId = setTimeout(() => {
            dispatch(
                addMessage(
                    chatId,
                    {
                        messageId: `id${Date.now()}`,
                        text: 'Ваше сообщение прочитано',
                        author: AUTHORS.BOT,
                    }
                )
            )
        }, 1500);
    }
}; */

export const updateMessages = (messages) => ({
    type: UPDATE_MESSAGES,
    payload: {
        messages,
    },
});