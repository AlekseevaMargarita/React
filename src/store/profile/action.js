import { onValue, set } from "firebase/database";
import { userNameRef, userShowNameRef } from "../../services/firebase";

export const TOGGLE_NAME = 'TOGGLE_NAME';
export const SET_NAME = 'SET_NAME';

export const toggleName = (value) => ({
    type: TOGGLE_NAME,
    payload: {
        value,
    },
});

export const setName = (name) => ({
    type: SET_NAME,
    payload: {
        name,
    },
});

export const initUserData = () => (dispatch) => {
    onValue(userNameRef, (snapshot) => {
        const userName = snapshot?.val();
        dispatch(setName(userName));
    });
    onValue(userShowNameRef, (snapshot) => {
        const userShowName = snapshot?.val();
        dispatch(setName(userShowName));
    });
};

export const setNameInDB = (newName) => () => {
    set(userNameRef, newName);
};

export const setShowNameInDB = (newValue) => () => {
    set(userShowNameRef, newValue);
};