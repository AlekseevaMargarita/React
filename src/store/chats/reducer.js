import { ADD_CHAT } from "./action"

const initialState = [];

export const chatsReducer = (state = initialState, action) => {
    switch (action.type) {
        case ADD_CHAT:
            return [
                ...state,
                action.payload.chat,
            ];
        /*         case DELETE_CHAT:
                    return [
                        ...state,
        
                    ] */
        default:
            return state;
    }
};
