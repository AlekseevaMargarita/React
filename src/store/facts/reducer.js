
import { STATUSES } from "../../constants/constants";
import { GET_FACTS_FAILURE, GET_FACTS_REQUEST, GET_FACTS_SUCCESS } from "./action";

const initialState = {
    facts: [],
    request: STATUSES.IDLE,
    error: null,
    loading: false,
};

export const factsReducer = (state = initialState, action) => {
    switch (action.type) {
        case GET_FACTS_REQUEST:
            return {
                ...state,
                facts: [],
                request: STATUSES.REQUEST,
                error: null,
                loading: true,

            };
        case GET_FACTS_SUCCESS:
            return {
                ...state,
                request: STATUSES.SUCCESS,
                loading: false,
                facts: action.payload,
            };
        case GET_FACTS_FAILURE:
            return {
                ...state,
                request: STATUSES.FAILURE,
                error: action.payload,
                loading: false,
            };
        default:
            return state;
    }
};