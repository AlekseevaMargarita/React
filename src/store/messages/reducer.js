import { UPDATE_MESSAGES } from "./action";

const initialState = {
    messages: {},
};

export const messagesReducer = (state = initialState, action) => {
    switch (action.type) {
        case UPDATE_MESSAGES:
            return {
                ...state,
                messages: action.payload.messages,
            };
        default:
            return state;
    }
};

