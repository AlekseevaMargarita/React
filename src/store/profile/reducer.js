import { SET_NAME, TOGGLE_NAME } from "./action";

const initialState = {
    showName: true,
    name: 'NoName',
};

export const profileReducer = (state = initialState, action) => {
    switch (action.type) {
        case TOGGLE_NAME:
            return {
                ...state,
                showName: action.payload.value,
            };
        case SET_NAME:
            return {
                ...state,
                name: action.payload.name,
            };
        default:
            return state;
    }
};