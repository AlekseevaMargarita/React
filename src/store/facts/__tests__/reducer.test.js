import { getFactsFailure, getFactsRequest, getFactsSuccess } from '../action';
import { factsReducer, initialState } from '../reducer';

describe('factsReducer tests with snapshot', () => {
    it('--- return correct state after GET_FACTS_REQUEST action', () => {
        const factStore = factsReducer(initialState, getFactsRequest());
        expect(factStore).toMatchSnapshot();
    });

    it('--- return correct state after GET_FACTS_SUCCESS action', () => {
        const factStore = factsReducer(initialState, getFactsSuccess());
        expect(factStore).toMatchSnapshot();
    });

    it('--- return correct state after GET_FACTS_FAILURE action', () => {
        const factStore = factsReducer(initialState, getFactsFailure());
        expect(factStore).toMatchSnapshot();
    });
});
