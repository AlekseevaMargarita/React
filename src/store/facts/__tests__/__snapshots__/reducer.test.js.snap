// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factsReducer tests with snapshot --- return correct state after GET_FACTS_FAILURE action 1`] = `
Object {
  "error": undefined,
  "facts": Array [],
  "loading": false,
  "request": 3,
}
`;

exports[`factsReducer tests with snapshot --- return correct state after GET_FACTS_REQUEST action 1`] = `
Object {
  "error": null,
  "facts": Array [],
  "loading": true,
  "request": 1,
}
`;

exports[`factsReducer tests with snapshot --- return correct state after GET_FACTS_SUCCESS action 1`] = `
Object {
  "error": null,
  "facts": undefined,
  "loading": false,
  "request": 2,
}
`;
